src/freenet/node/PeerLocation.java

```diff
@@ -56,43 +60,43 @@ public synchronized int getDegree() {
 	}
 
 	boolean updateLocation(double newLoc, double[] newLocs) {
-		if(!Location.isValid(newLoc)) {
+		if (!Location.isValid(newLoc)) {
 			Logger.error(this, "Invalid location update for " + this+ " ("+newLoc+')', new Exception("error"));
 			// Ignore it
 			return false;
 		}
 
-		for(double currentLoc : newLocs) {
-			if(!Location.isValid(currentLoc)) {
-				Logger.error(this, "Invalid location update for " + this + " ("+currentLoc+')', new Exception("error"));
+		final double[] newPeersLocation = new double[newLocs.length];
+		for (int i = 0; i < newLocs.length; i++) {
+			final double loc = newLocs[i];
+			if (!Location.isValid(loc)) {
+				Logger.error(this, "Invalid location update for " + this + " (" + loc + ")", new Exception("error"));
 				// Ignore it
 				return false;
```

Comment:
this is not a change in behaviour, so I think it should not be touched in the complexity improvement. changing `return false` here would entangle changes which should purely improve performance with changes which cause behaviour change — which are two changes which I think should not be released together.